### ISSUE TITLE
Anthropic to OpenAI translation silently drops thinking.type=adaptive and returns 200

### ISSUE THREAD

**Issue by Atharva-Kanherkar** [ogx-ai/ogx#6614]
### System Info

```
OGX v1.4.0, commit 051a8a0
Python 3.12 (also observed on 3.13)
macOS 26.2, arm64
Backend: OpenAI-compatible provider (translation mode)
```

This is a request translation defect in the Anthropic compatibility layer. It does
not involve a GPU or a model runtime, so `torch.utils.collect_env` output would not
be meaningful here.

### Information

- [x] My own modified scripts

### Describe the bug

`ThinkingConfig.type` accepts three variants, but the Anthropic to OpenAI
translation guard only handles one of them.

`src/ogx_api/messages/models.py:240`

```python
type: Literal["enabled", "disabled", "adaptive"] = "enabled"
```

`src/ogx/providers/utils/inference/anthropic_translation.py:234`

```python
def anthropic_request_to_openai(request: AnthropicCreateMessageRequest) -> OpenAIChatCompletionRequestWithExtraBody:
    """Convert an Anthropic CreateMessage request to OpenAI chat completion params."""
    if request.thinking and request.thinking.type == "enabled":
        raise ValueError(
            "Failed to process thinking request: extended thinking requires a native "
            "Anthropic-compatible provider; translation mode does not support it"
        )
```

`request.thinking` is never read again in that function, so any variant other than
`enabled` is discarded without being mapped to a target dialect field:

| `thinking.type` | Behavior | Correct |
|---|---|---|
| `enabled` | rejected, HTTP 400 | yes |
| `disabled` | dropped | yes, there is nothing to forward |
| `adaptive` | dropped, HTTP 200 | no, the caller asked for a reasoning policy and is told it succeeded |

`adaptive` is not a malformed input. Anthropic's own Messages API accepts the same
object, so an Anthropic SDK client pointed at OGX can legitimately send it.

#### Minimal reproduction, no server required

```python
from ogx.providers.utils.inference.anthropic_translation import anthropic_request_to_openai
from ogx_api.messages.models import AnthropicCreateMessageRequest


def build(thinking):
    return AnthropicCreateMessageRequest(
        model="mock-gpt",
        max_tokens=32,
        messages=[{"role": "user", "content": "hi"}],
        thinking=thinking,
    )


for variant in ({"type": "enabled", "budget_tokens": 1024}, {"type": "adaptive"}):
    req = build(variant)
    try:
        out = anthropic_request_to_openai(req)
    except ValueError as exc:
        print(f"{variant['type']:>8}: rejected -> {exc}")
        continue
    forwarded = out.model_dump(exclude_none=True)
    reasoning_keys = [k for k in forwarded if "reason" in k or "think" in k]
    print(f"{variant['type']:>8}: accepted -> forwarded reasoning keys: {reasoning_keys}")
```

Output on v1.4.0:

```
 enabled: rejected -> Failed to process thinking request: extended thinking requires a native Anthropic-compatible provider; translation mode does not support it
adaptive: accepted -> forwarded reasoning keys: []
```

#### End to end over the public route

Sending this to `POST /v1/messages` with an OpenAI-compatible backend:

```json
{
  "model": "openai/mock-gpt",
  "max_tokens": 32,
  "thinking": {"type": "adaptive"},
  "messages": [{"role": "user", "content": "Synthetic adaptive-thinking control."}]
}
```

produces this upstream request, with no reasoning configuration of any kind:

```json
{
  "messages": [{"role": "user", "content": "Synthetic adaptive-thinking control."}],
  "model": "mock-gpt",
  "max_completion_tokens": 32,
  "stream": false
}
```

and the client receives HTTP 200 with a normal assistant message. Reproduced 5 out
of 5 trials. The same route with only `thinking.type` changed to `enabled` returns
HTTP 400 in 5 out of 5 trials and sends no upstream request, which isolates the
difference to the union variant rather than the route, model, or transport.

### Error logs

There is no error or traceback, and that is the defect. The request succeeds:

```
HTTP/1.1 200 OK

{"id": "msg_...", "type": "message", "role": "assistant",
 "content": [{"type": "text", "text": "Synthetic control response."}],
 "model": "mock-gpt", "stop_reason": "end_turn",
 "usage": {"input_tokens": 1, "output_tokens": 1}}
```

For contrast, the `enabled` variant on the same route returns the intended
fail-closed error:

```
HTTP/1.1 400 Bad Request

{"type": "error", "error": {"type": "invalid_request_error",
 "message": "Failed to process thinking request: extended thinking requires a native Anthropic-compatible provider; translation mode does not support it"}}
```

Because the failure is silent, a client cannot detect the loss, cannot fall back to
a native Anthropic provider, and cannot surface the missing constraint to a user.

### Expected behavior

Either of these would be consistent. I would suggest the first, since it matches the
behavior already chosen for `enabled`:

1. Reject `adaptive` in translation mode with the same error. The existing message
   already states the general rule, that translation mode does not support extended
   thinking, which applies to `adaptive` just as it does to `enabled`. The fix is to
   widen the guard to the variants that carry a reasoning request:

   ```python
   if request.thinking and request.thinking.type in ("enabled", "adaptive"):
   ```

   `disabled` correctly requires no action, since there is nothing to forward.

2. Map `adaptive` to an explicit target dialect reasoning configuration, for example
   an OpenAI `reasoning_effort`, when the backend supports one.

What should not happen is accepting the field, discarding it, and reporting success.

Note that the surrounding code in `anthropic_request_to_openai` is careful to
preserve other request constraints, including `top_k`, `stop_sequences`,
`service_tier`, and `disable_parallel_tool_use` mapped to `parallel_tool_calls`.
`thinking` appears to be an omission rather than a deliberate exclusion.

I am happy to open a PR for option 1 with a test alongside the existing
`enabled` coverage.
